savoirfairelinux/sous-chef
# Fix #601 (route=None in kitchen count)

## Fix #601 by lingxiaoyang

### Changes proposed in this pull request:

* Consider when `client.route is None`, when sorting the generated orders.
* If a client has null route, the corresponding field will show a warning sign, linking to client update form. (originally, the warning sign is shown when a client is not geolocalized.)
![image](https://cloud.githubusercontent.com/assets/8630726/21508527/d0ca9a34-cc4f-11e6-8b66-02823a0b6acf.png)

* New unit test of the display of warning signs.
* New unit test of the entire kitchen count procedure, to test if `route=None` clients are excluded in each step.
* Fix the problem that `route=None` clients are not excluded in `clashing ingredients` table.
* Also exclude all non-geolocalized clients in next steps of the kitchen count.


### Status


### How to verify this change

1. Generate today's delivery orders.
2. Choose a client, go to Django admin, set route to null.
3. Follow the steps in kitchen count. This client should now be excluded in the procedure.
4. For ungeolocalized clients (client.member.address.latitude/longitude is None), there's a warning in organize route page, but they are included in the procedure.

### Deployment notes and migration

none

### New translatable strings


### Additional notes

none
